### PR TITLE
feat: [IOPLT-384] Removes Mixpanel community library to use the official npm package

### PR DESCRIPTION
**Pull request by CrisTofani**
## Short description
This PR want to remove the legacy Mixpanel community library and replaces it with the `mixpanel-react-native` npm package 

> [!note]
> Push notification initialization for Mixpanel has been removed since it is actually not supported by the react-native wrapper and is planned to be deprecated by native-sdk as per https://github.com/mixpanel/mixpanel-react-native/issues/89#issuecomment-904192090

## List of changes proposed in this pull request
- Changes the library
- Remap the initializagtion file
- Fix compilation issues due to the change of library

## How to test
Chack mixpanel events are correctly dispatched and tracked

### CODE REVIEW

**Review by shadowsheep1 — CHANGES_REQUESTED** [review]
On iOS side I have those not committed changes.

[mixpanel.patch](https://github.com/pagopa/io-app/files/14755035/mixpanel.patch)

**Comment by CrisTofani** [comment]
> On iOS side I have those not committed changes.
> 
> [mixpanel.patch](https://github.com/pagopa/io-app/files/14755035/mixpanel.patch)

Great catch @shadowsheep1 ! 

They were the file from the old version of MP I forgot to remove on iOS side, now should be everything fine.

commited in [e4ceaa7](https://github.com/pagopa/io-app/pull/5601/commits/e4ceaa747c92f6856e035463a8b15aa99bb7c02f)

File: ts/mixpanel.ts [hunk 1]
```diff
@@ -16,22 +15,22 @@ export const initializeMixPanel = async (state: GlobalState) => {
...
     return;
   }
-  const privateInstance = new MixpanelInstance(mixpanelToken);
-  await privateInstance.initialize();
+  const trackAutomaticEvents = false;
+  const privateInstance = new Mixpanel(mixpanelToken, trackAutomaticEvents);
+  await privateInstance.init(
+    undefined,
+    undefined,
+    "https://api-eu.mixpanel.com/"
```
Comment (shadowsheep1) [inline ts/mixpanel.ts:23]:
```suggestion
    "https://api-eu.mixpanel.com"
```
I've found that with the trailing slash we have a `301` status code from mixpanel.

image

This lead to a missing event tracking.

[the remark this answers is not in the document]
Comment (CrisTofani) [reply ts/mixpanel.ts]: We just patched the community sdk to track directly on the eu-server, we can for sure move this in the config file.

**Comment by shadowsheep1** [comment]
Build Android e tracking ok ✅

**Review by shadowsheep1 — APPROVED** [review]
LGTM 🚀

**Pull request merged**